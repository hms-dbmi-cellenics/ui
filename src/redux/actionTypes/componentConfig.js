/**
 * Sets plot configs and data.
 */
const COMPONENT_CONFIG = 'componentConfig';
const LOAD_CONFIG = `${COMPONENT_CONFIG}/load`;
const UPDATE_CONFIG = `${COMPONENT_CONFIG}/update`;
const RESET_CONFIG = `${COMPONENT_CONFIG}/reset`;
const SAVE_CONFIG = `${COMPONENT_CONFIG}/save`;
const PLOT_DATA_LOADED = `${COMPONENT_CONFIG}/plotDataLoaded`;
const PLOT_DATA_LOADING = `${COMPONENT_CONFIG}/plotDataLoading`;
const PLOT_DATA_ERROR = `${COMPONENT_CONFIG}/plotDataError`;

export {
  LOAD_CONFIG, UPDATE_CONFIG, RESET_CONFIG, SAVE_CONFIG,
  PLOT_DATA_LOADED, PLOT_DATA_LOADING, PLOT_DATA_ERROR,
};
