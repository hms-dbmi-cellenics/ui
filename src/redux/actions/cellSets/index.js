import createCellSet from './createCellSet';
import deleteCellSet from './deleteCellSet';
import deleteCellClass from './deleteCellClass';
import runCellSetsClustering from './runCellSetsClustering';
import runCellSetsAnnotation from './runCellSetsAnnotation';
import runCellCycleScoring from './runCellCycleScoring';
import updateCellSetsClustering from './updateCellSetsClustering';

import loadCellSets from './loadCellSets';
import reorderCellSet from './reorderCellSet';
import updateCellSetProperty from './updateCellSetProperty';
import updateCellSetSelected from './updateCellSetSelected';
import setCellSetHiddenStatus from './setCellSetHiddenStatus';
import unhideAllCellSets from './unhideAllCellSets';

export {
  createCellSet,
  deleteCellSet,
  deleteCellClass,
  runCellSetsClustering,
  runCellSetsAnnotation,
  updateCellSetsClustering,
  loadCellSets,
  reorderCellSet,
  updateCellSetSelected,
  updateCellSetProperty,
  setCellSetHiddenStatus,
  unhideAllCellSets,
  runCellCycleScoring,
};
