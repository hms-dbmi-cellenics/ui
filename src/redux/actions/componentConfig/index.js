import loadPlotConfig from 'redux/actions/componentConfig/loadPlotConfig';
import loadComponentConfig from 'redux/actions/componentConfig/loadComponentConfig';
import updatePlotConfig from 'redux/actions/componentConfig/updatePlotConfig';
import savePlotConfig from 'redux/actions/componentConfig/savePlotConfig';
import fetchPlotDataWork from 'redux/actions/componentConfig/fetchPlotDataWork';
import updatePlotData from 'redux/actions/componentConfig/updatePlotData';
import resetPlotConfig from 'redux/actions/componentConfig/resetPlotConfig';

export {
  loadPlotConfig,
  loadComponentConfig,
  updatePlotConfig,
  savePlotConfig,
  fetchPlotDataWork,
  updatePlotData,
  resetPlotConfig,
};
