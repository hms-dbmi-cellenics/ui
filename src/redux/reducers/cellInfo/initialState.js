const initialState = {
  component: null,
  cellId: null,
  geneName: null,
  trackCluster: null,
  focus: {
    store: 'cellSets',
    key: 'louvain',
  },
  groupedTrack: 'louvain',
  selectedTracks: ['louvain'],
};

export default initialState;
