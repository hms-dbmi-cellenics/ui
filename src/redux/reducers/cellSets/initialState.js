const initialState = {
  properties: {},
  hierarchy: [],
  selected: {},
  initialLoadPending: true,
  loading: false,
  error: false,
  updatingClustering: false,
  hidden: new Set(),
};

export default initialState;
