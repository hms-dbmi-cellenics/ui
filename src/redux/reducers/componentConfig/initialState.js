import { plotTypes } from 'utils/constants';

import {
  legendBaseState,
  dimensionsBaseState,
  axesBaseState,
  axesRangesBaseState,
  titleBaseState,
  fontStyleBaseState,
  colourBaseState,
  markerBaseState,
  labelBaseState,
} from './baseStylesState';

// PLOTS & TABLES - Categorical Embedding
const embeddingCategoricalInitialConfig = {
  spec: '1.0.0',
  legend: {
    ...legendBaseState,
  },
  dimensions: {
    ...dimensionsBaseState,
    width: 700,
    height: 550,
  },
  axes: {
    ...axesBaseState,
    xAxisText: 'Umap 1',
    yAxisText: 'Umap 2',
    defaultValues: ['x', 'y'],
    offset: 0,
  },
  axesRanges: axesRangesBaseState,
  title: {
    ...titleBaseState,
    fontSize: 20,
  },
  fontStyle: fontStyleBaseState,
  colour: colourBaseState,
  marker: markerBaseState,
  labels: {
    ...labelBaseState,
    enabled: false,
  },
  selectedCellSet: 'louvain',
  selectedSample: 'All',
};

// PLOTS & TABLES - Continuous Embedding
const embeddingContinuousInitialConfig = {
  spec: '1.0.0',
  legend: legendBaseState,
  dimensions: {
    ...dimensionsBaseState,
    width: 700,
    height: 550,
  },
  axes: {
    ...axesBaseState,
    xAxisText: 'Umap 1',
    yAxisText: 'Umap 2',
    defaultValues: ['x', 'y'],
    offset: 10,
  },
  axesRanges: axesRangesBaseState,
  title: {
    ...titleBaseState,
    dx: 0,
    fontSize: 20,
  },
  fontStyle: fontStyleBaseState,
  colour: colourBaseState,
  marker: markerBaseState,
  labels: labelBaseState,
  logEquation: 'datum.expression*1',
  shownGene: null,
  expressionValue: 'raw',
  truncatedValues: true,
  selectedSample: 'All',
  keepValuesOnReset: ['shownGene'],
};

// PLOTS & TABLES - Heatmap
const heatmapInitialConfig = {
  spec: '1.0.0',
  legend: {
    ...legendBaseState,
    show: true,
    position: 'horizontal',
  },
  dimensions: {
    ...dimensionsBaseState,
    width: 500,
    height: 500,
  },
  title: {
    ...titleBaseState,
    fontSize: 20,
    dx: 0,
  },
  fontStyle: fontStyleBaseState,
  colour: colourBaseState,
  marker: markerBaseState,
  labels: labelBaseState,
  // When null, this indicates that the value is its initial one
  // and the marker genes should be loaded
  selectedGenes: null,
  selectedCellSet: 'louvain',
  selectedPoints: 'All',
  labelColour: 'transparent',
  selectedTracks: ['louvain'],
  groupedTracks: ['sample', 'louvain'],
  expressionValue: 'raw',
  truncatedValues: true,
};

// PLOTS & TABLES - Marker heatmap
const markerHeatmapInitialConfig = {
  ...heatmapInitialConfig,
  useMarkerGenes: false,
  guardLines: false,
  nMarkerGenes: 5,
  showGeneLabels: true,
  keepValuesOnReset: ['selectedGenes'],
};

// PLOTS & TABLES - Volcano plot
const volcanoInitialConfig = {
  spec: '1.0.0',
  legend: {
    ...legendBaseState,
    position: 'bottom-right',
  },
  dimensions: dimensionsBaseState,
  marker: {
    ...markerBaseState,
    showOpacity: false,
    size: 32,
  },
  axes: {
    ...axesBaseState,
    xAxisText: 'log fold change',
    yAxisText: '-log10(adj p-value)',
    gridOpacity: 5,
    offset: 10,
  },
  axesRanges: axesRangesBaseState,
  title: titleBaseState,
  fontStyle: fontStyleBaseState,
  colour: colourBaseState,
  labels: labelBaseState,
  noDifferenceColor: '#aaaaaa',
  significantUpregulatedColor: '#0000ffaa',
  significantDownregulatedColor: '#ff0000',

  xAxisAuto: true,
  yAxisAuto: true,
  logFoldChangeDomain: 1,
  maxNegativeLogpValueDomain: 50,
  negLogpValueThreshold: 4,
  logFoldChangeThreshold: 1,
  logFoldChangeTickCount: 5,
  negativeLogpValueTickCount: 5,
  downsampleRatio: 0,
  showLogFoldChangeThresholdGuides: true,
  showpvalueThresholdGuides: true,
  thresholdGuideWidth: 1,
  logFoldChangeThresholdColor: '#ff0000',
  pvalueThresholdColor: '#ff0000',
  textThresholdValue: 240,
  strokeOpa: 1,
  strokeCol: '#000000',
};

// PLOTS & TABLES - Frequency
const frequencyInitialConfig = {
  spec: '1.0.0',
  frequencyType: 'proportional',
  legend: {
    ...legendBaseState,
    title: 'Cell Set',
    offset: 40,
  },
  labels: labelBaseState,
  dimensions: dimensionsBaseState,
  marker: markerBaseState,
  colour: colourBaseState,
  title: titleBaseState,
  axes: {
    ...axesBaseState,
    xAxisText: 'Sample',
    yAxisText: 'Proportion',
    xAxisRotateLabels: true,
    offset: 10,
  },
  axesRanges: {
    yAxisAuto: true,
    yMin: 0,
    yMax: 10,
  },
  fontStyle: fontStyleBaseState,
  proportionGrouping: 'louvain',
  xAxisGrouping: 'sample',
  axisTitlesize: 13,
  geneexpLegendloc: '',
};

// PLOTS & TABLES - Violin
const violinConfig = {
  spec: '1.0.0',
  legend: {
    ...legendBaseState,
    enabled: false,
  },
  dimensions: {
    ...dimensionsBaseState,
    width: 700,
    height: 550,
  },
  axes: {
    ...axesBaseState,
    offset: 10,
    xAxisRotateLabels: true,
  },
  axesRanges: {
    yAxisAuto: true,
    yMin: 0,
    yMax: 10,
  },
  title: {
    ...titleBaseState,
    dx: 0,
    fontSize: 20,
  },
  fontStyle: fontStyleBaseState,
  colour: colourBaseState,
  marker: markerBaseState,
  labels: labelBaseState,
  shownGene: 'notSelected',
  selectedCellSet: 'louvain',
  selectedPoints: 'All',
  selectedPointsVisible: true,
  statisticsVisible: false,
  kdeBandwidth: 0.3,
  normalised: 'zScore',
  keepValuesOnReset: ['shownGene', 'title'],
};

// PLOTS & TABLES - Dot Plot
const dotPlotConfig = {
  spec: '1.0.0',
  legend: {
    ...legendBaseState,
    enabled: true,
    direction: 'vertical',
  },
  dimensions: {
    ...dimensionsBaseState,
    width: 700,
    height: 550,
  },
  axes: {
    ...axesBaseState,
    tickOffset: 10,
    xAxisRotateLabels: true,
    xAxisText: 'Gene names',
    yAxisText: 'Louvain clusters',
  },
  title: {
    ...titleBaseState,
    dx: 0,
    fontSize: 20,
  },
  fontStyle: fontStyleBaseState,
  colour: colourBaseState,
  marker: markerBaseState,
  labels: labelBaseState,
  useAbsoluteScale: true,
  useMarkerGenes: false,
  nMarkerGenes: 3,
  selectedGenes: [],
  selectedCellSet: 'louvain',
  selectedPoints: 'All',
  keepValuesOnReset: ['selectedGenes'],
};

// PLOTS & TABLES - Trajectory Analysis
const trajectoryAnalysisInitialConfig = {
  spec: '1.0.0',
  legend: {
    ...legendBaseState,
  },
  dimensions: {
    ...dimensionsBaseState,
    width: 700,
    height: 550,
  },
  axes: {
    ...axesBaseState,
    xAxisText: 'Umap 1',
    yAxisText: 'Umap 2',
    defaultValues: ['x', 'y'],
  },
  axesRanges: axesRangesBaseState,
  title: {
    ...titleBaseState,
    fontSize: 20,
  },
  fontStyle: fontStyleBaseState,
  colour: colourBaseState,
  marker: markerBaseState,
  labels: {
    ...labelBaseState,
    enabled: false,
  },
  selectedNodes: [],
  selectedCellSets: ['louvain'],
  embeddingCellSet: 'louvain',
  embeddingSample: 'All',
};

// PLOTS & TABLES - Multi view
const multiViewInitialConfig = {
  ncols: 1,
  nrows: 1,
  plotUuids: [],
};

// EMBEDDING PREVIEW - Coloured by sample
const embeddingPreviewBySampleInitialConfig = {
  spec: '1.0.0',
  legend: {
    ...legendBaseState,
  },
  dimensions: {
    ...dimensionsBaseState,
    width: 700,
    height: 550,
  },
  axes: {
    ...axesBaseState,
    xAxisText: '',
    yAxisText: '',
    defaultValues: ['x', 'y'],
    offset: 10,
  },
  axesRanges: axesRangesBaseState,
  title: {
    ...titleBaseState,
    fontSize: 20,
  },
  fontStyle: fontStyleBaseState,
  colour: colourBaseState,
  marker: markerBaseState,
  labels: {
    ...labelBaseState,
    enabled: false,
  },
  selectedCellSet: 'louvain',
  selectedSample: 'All',
};

// EMBEDDING PREVIEW - Coloured by cell sets
const embeddingPreviewByCellSetsInitialConfig = {
  spec: '1.0.0',
  legend: {
    ...legendBaseState,
  },
  dimensions: {
    ...dimensionsBaseState,
    width: 700,
    height: 550,
  },
  axes: {
    ...axesBaseState,
    xAxisText: '',
    yAxisText: '',
    defaultValues: ['x', 'y'],
    offset: 10,
  },
  axesRanges: axesRangesBaseState,
  title: {
    ...titleBaseState,
    fontSize: 20,
  },
  fontStyle: fontStyleBaseState,
  colour: colourBaseState,
  marker: markerBaseState,
  labels: {
    ...labelBaseState,
    enabled: false,
  },
  selectedCellSet: 'louvain',
  selectedSample: 'All',
};

// EMBEDDING PREVIEW - Config for fraction of mitochondrial reads
const embeddingPreviewMitochondrialContentInitialConfig = {
  spec: '1.0.0',
  legend: legendBaseState,
  dimensions: {
    ...dimensionsBaseState,
    width: 700,
    height: 550,
  },
  axes: {
    ...axesBaseState,
    xAxisText: '',
    yAxisText: '',
    defaultValues: ['x', 'y'],
    offset: 10,
  },
  axesRanges: axesRangesBaseState,
  title: {
    ...titleBaseState,
    dx: 0,
    fontSize: 20,
  },
  fontStyle: fontStyleBaseState,
  colour: colourBaseState,
  marker: markerBaseState,
  labels: labelBaseState,
  shownGene: 'notSelected',
  selectedSample: 'All',
};

// EMBEDDING PREVIEW - Config for doublet score
const embeddingPreviewDoubletScoreInitialConfig = {
  spec: '1.0.0',
  legend: legendBaseState,
  dimensions: {
    ...dimensionsBaseState,
    width: 700,
    height: 550,
  },
  axes: {
    ...axesBaseState,
    xAxisText: '',
    yAxisText: '',
    defaultValues: ['x', 'y'],
    offset: 10,
  },
  axesRanges: axesRangesBaseState,
  title: {
    ...titleBaseState,
    dx: 0,
    fontSize: 20,
  },
  fontStyle: fontStyleBaseState,
  colour: colourBaseState,
  marker: markerBaseState,
  labels: labelBaseState,
  selectedSample: 'All',
};

const embeddingPreviewNumOfGenesInitialConfig = {
  spec: '1.0.0',
  legend: legendBaseState,
  dimensions: {
    ...dimensionsBaseState,
    width: 700,
    height: 550,
  },
  axes: {
    ...axesBaseState,
    xAxisText: '',
    yAxisText: '',
    defaultValues: ['x', 'y'],
    offset: 10,
  },
  axesRanges: axesRangesBaseState,
  title: {
    ...titleBaseState,
    dx: 0,
    fontSize: 20,
  },
  fontStyle: fontStyleBaseState,
  colour: colourBaseState,
  marker: markerBaseState,
  labels: labelBaseState,
  selectedSample: 'All',
};
const embeddingPreviewNumOfUmisInitialConfig = {
  spec: '1.0.0',
  legend: legendBaseState,
  dimensions: {
    ...dimensionsBaseState,
    width: 700,
    height: 550,
  },
  axes: {
    ...axesBaseState,
    xAxisText: '',
    yAxisText: '',
    defaultValues: ['x', 'y'],
    offset: 10,
  },
  axesRanges: axesRangesBaseState,
  title: {
    ...titleBaseState,
    dx: 0,
    fontSize: 20,
  },
  fontStyle: fontStyleBaseState,
  colour: colourBaseState,
  marker: markerBaseState,
  labels: labelBaseState,
  selectedSample: 'All',
};

const interactiveHeatmapInitialConfig = {
  selectedCellSet: 'louvain',
  selectedPoints: 'All',
  selectedTracks: ['louvain'],
  groupedTracks: ['louvain', 'sample'],
  expressionValue: 'raw',
  legendIsVisible: true,
};

// CELL SIZE DISTRIBUTION - Cell Size Distribution Histogram
const cellSizeDistributionHistogram = {
  spec: '1.0.0',
  legend: {
    ...legendBaseState,
  },
  dimensions: {
    ...dimensionsBaseState,
    width: 530,
    height: 400,
  },
  axes: {
    ...axesBaseState,
    xAxisText: '#UMIs in cell',
    yAxisText: '#UMIs * #Cells',
  },
  axesRanges: axesRangesBaseState,
  title: {
    ...titleBaseState,
    fontSize: 20,
  },
  fontStyle: fontStyleBaseState,
  colour: colourBaseState,
  marker: markerBaseState,
  labels: labelBaseState,
  minCellSize: 10800,
};

// CELL SIZE DISTRIBUTION - Cell Size Distribution Knee Plot
const cellSizeDistributionKneePlot = {
  spec: '1.0.0',
  legend: {
    ...legendBaseState,
  },
  dimensions: {
    ...dimensionsBaseState,
    width: 530,
    height: 400,
  },
  axes: {
    ...axesBaseState,
    xAxisText: 'Cell rank',
    yAxisText: '#UMIs in cell',
  },
  axesRanges: {
    ...axesRangesBaseState,
    xMin: 1,
    yMin: 1,
  },
  title: {
    ...titleBaseState,
    fontSize: 20,
  },
  fontStyle: fontStyleBaseState,
  colour: colourBaseState,
  marker: markerBaseState,
  labels: labelBaseState,
  minCellSize: 990,
};

// MITOCHONDRIAL CONTENT - Mitochondrial Fraction Histogram
const mitochondrialFractionHistogram = {
  spec: '1.0.0',
  legend: {
    ...legendBaseState,
  },
  dimensions: {
    ...dimensionsBaseState,
    width: 530,
    height: 400,
  },
  axes: {
    ...axesBaseState,
    xAxisText: 'Percentage of mitochondrial reads',
    yAxisText: 'Percentage of cells',
  },
  axesRanges: axesRangesBaseState,
  title: {
    ...titleBaseState,
    fontSize: 20,
    dx: 0,
  },
  fontStyle: fontStyleBaseState,
  colour: colourBaseState,
  marker: markerBaseState,
  labels: labelBaseState,
  maxFraction: 0.1,
};

// MITOCHONDRIAL CONTENT - Mitochondrial Fraction Log Histogram
const mitochondrialFractionLogHistogram = {
  spec: '1.0.0',
  legend: {
    ...legendBaseState,
  },
  dimensions: {
    ...dimensionsBaseState,
    width: 530,
    height: 400,
  },
  axes: {
    ...axesBaseState,
    xAxisText: 'Percentage of mitochondrial reads',
    yAxisText: 'Num of UMIs in cell',
  },
  axesRanges: axesRangesBaseState,
  title: {
    ...titleBaseState,
    fontSize: 20,
    dx: 0,
  },
  fontStyle: fontStyleBaseState,
  colour: colourBaseState,
  marker: markerBaseState,
  labels: labelBaseState,
  maxFraction: 0.2,
};

// CLASSIFIER - Classifier Knee Plot
const classifierKneePlot = {
  spec: '1.0.0',
  legend: {
    ...legendBaseState,
  },
  dimensions: {
    ...dimensionsBaseState,
    width: 530,
    height: 400,
  },
  axes: {
    ...axesBaseState,
    xAxisText: 'Droplet Rank',
    yAxisText: 'Droplet #UMIs',
  },
  axesRanges: {
    ...axesRangesBaseState,
    xMin: 1,
    yMin: 1,
  },
  title: {
    ...titleBaseState,
    fontSize: 20,
  },
  fontStyle: fontStyleBaseState,
  colour: colourBaseState,
  marker: markerBaseState,
  labels: labelBaseState,
  minCellSize: 990,
};

// CLASSIFIER - Classifier Empty Drops
const classifierEmptyDropsPlot = {
  spec: '1.0.0',
  legend: {
    ...legendBaseState,
    enabled: false,
  },
  dimensions: {
    ...dimensionsBaseState,
    width: 630,
    height: 500,
  },
  axes: {
    ...axesBaseState,
    xAxisText: 'log10[ cell size (UMIs) ]',
    yAxisText: 'FDR (emptyDrops)',
    gridOpacity: 10,
  },
  axesRanges: axesRangesBaseState,
  title: {
    ...titleBaseState,
    fontSize: 20,
  },
  fontStyle: fontStyleBaseState,
  colour: colourBaseState,
  marker: markerBaseState,
  labels: labelBaseState,
  minProbability: 0.82,
  bandwidth: -1,
};

// GENES VS UMIS - Features vs UMIs
const featuresVsUMIsScatterplot = {
  spec: '1.0.0',
  dimensions: {
    ...dimensionsBaseState,
    width: 630,
    height: 500,
  },
  axes: {
    ...axesBaseState,
    xAxisText: 'log10 [molecule counts]',
    yAxisText: 'log10 [gene counts]',
    gridOpacity: 10,
  },
  axesRanges: axesRangesBaseState,
  title: {
    ...titleBaseState,
    fontSize: 20,
  },
  fontStyle: fontStyleBaseState,
  colour: colourBaseState,
  labels: labelBaseState,
  lower_cutoff: 4.8,
  upper_cutoff: 2.1,
};

// DOUBLE FILTER - Doublet filter histogram
const doubletScoreHistogram = {
  spec: '1.0.0',
  legend: {
    ...legendBaseState,
  },
  dimensions: {
    ...dimensionsBaseState,
    width: 530,
    height: 400,
  },
  axes: {
    ...axesBaseState,
    xAxisText: 'Probability of being a doublet',
    yAxisText: 'Frequency',
  },
  axesRanges: axesRangesBaseState,
  title: {
    ...titleBaseState,
    fontSize: 20,
  },
  fontStyle: fontStyleBaseState,
  colour: colourBaseState,
  marker: markerBaseState,
  labels: labelBaseState,
  probThreshold: 0.2,
};

// DATA INTEGRATION - Embedding by Samples
const dataIntegrationEmbeddingInitialConfig = {
  spec: '1.0.0',
  legend: {
    ...legendBaseState,
  },
  dimensions: {
    ...dimensionsBaseState,
    width: 700,
    height: 550,
  },
  axes: {
    ...axesBaseState,
    xAxisText: '',
    yAxisText: '',
    defaultValues: ['x', 'y'],
    offset: 10,
  },
  axesRanges: axesRangesBaseState,
  title: {
    ...titleBaseState,
    fontSize: 20,
  },
  fontStyle: fontStyleBaseState,
  colour: colourBaseState,
  marker: markerBaseState,
  labels: {
    ...labelBaseState,
    enabled: false,
  },
  selectedCellSet: 'louvain',
  selectedSample: 'All',
};

// DATA INTEGRATION - Frequency
const dataIntegrationFrequencyInitialConfig = {
  ...frequencyInitialConfig,
  xAxisGrouping: 'louvain',
  proportionGrouping: 'sample',
  legend: {
    ...legendBaseState,
    title: 'Sample',
    offset: 10,
  },
  dimensions: {
    ...dimensionsBaseState,
    width: 700,
    height: 550,
  },

  axes: {
    ...axesBaseState,
    xAxisText: 'Louvain clusters',
    yAxisText: 'Proportion',
    xAxisRotateLabels: true,
    offset: 10,
  },
  axesRanges: {
    yAxisAuto: true,
    yMin: 0,
    yMax: 10,
  },
  fontStyle: fontStyleBaseState,
};

// DATA INTEGRATION - Elbow
const dataIntegrationElbowPlotInitialConfig = {
  legend: {
    ...legendBaseState,
  },
  labels: labelBaseState,
  dimensions: {
    ...dimensionsBaseState,
    width: 700,
    height: 550,
  },
  marker: { ...markerBaseState },
  fontStyle: fontStyleBaseState,
  axes: {
    ...axesBaseState,
    xAxisText: 'Principal Components',
    yAxisText: 'Proportion of Variance Explained',
    titleFont: 'sans-serif',
    labelFont: 'sans-serif',
    labelFontSize: 13,
    offset: 0,
    gridOpacity: 10,
  },
  axesRanges: axesRangesBaseState,
  colour: {
    toggleInvert: '#FFFFFF',
  },
  title: {
    ...titleBaseState,
    font: 'sans-serif',
    fontSize: 12,
  },
  signals: [
    {
      name: 'interpolate',
      value: 'linear',
      bind: {
        input: 'select',
        options: [
          'basis',
          'cardinal',
          'catmull-rom',
          'linear',
          'monotone',
          'natural',
          'step',
          'step-after',
          'step-before',
        ],
      },
    },
  ],
};

const normalizedMatrixPlotConfig = {
  sample: [],
  louvain: [],
  metadata: [],
  scratchpad: [],
};

const initialPlotConfigStates = {
  cellSizeDistributionHistogram,
  cellSizeDistributionKneePlot,
  mitochondrialFractionHistogram,
  mitochondrialFractionLogHistogram,
  classifierKneePlot,
  classifierEmptyDropsPlot,
  featuresVsUMIsScatterplot,
  doubletScoreHistogram,
  embeddingCategorical: embeddingCategoricalInitialConfig,
  embeddingContinuous: embeddingContinuousInitialConfig,
  heatmap: heatmapInitialConfig,
  volcano: volcanoInitialConfig,
  markerHeatmap: markerHeatmapInitialConfig,
  violin: violinConfig,
  [plotTypes.NORMALIZED_EXPRESSION_MATRIX]: normalizedMatrixPlotConfig,
  [plotTypes.DOT_PLOT]: dotPlotConfig,
  [plotTypes.TRAJECTORY_ANALYSIS]: trajectoryAnalysisInitialConfig,
  frequency: frequencyInitialConfig,
  embeddingPreviewBySample: embeddingPreviewBySampleInitialConfig,
  embeddingPreviewByCellSets: embeddingPreviewByCellSetsInitialConfig,
  embeddingPreviewMitochondrialContent: embeddingPreviewMitochondrialContentInitialConfig,
  embeddingPreviewDoubletScore: embeddingPreviewDoubletScoreInitialConfig,
  embeddingPreviewNumOfGenes: embeddingPreviewNumOfGenesInitialConfig,
  embeddingPreviewNumOfUmis: embeddingPreviewNumOfUmisInitialConfig,
  dataIntegrationEmbedding: dataIntegrationEmbeddingInitialConfig,
  dataIntegrationFrequency: dataIntegrationFrequencyInitialConfig,
  dataIntegrationElbow: dataIntegrationElbowPlotInitialConfig,
};

const initialComponentConfigStates = {
  interactiveHeatmap: interactiveHeatmapInitialConfig,
  multiView: multiViewInitialConfig,
};

const initialPlotDataState = {
  plotData: [],
  loading: false,
  error: false,
};

export {
  initialPlotConfigStates,
  initialComponentConfigStates,
  initialPlotDataState,
};

const initialState = {};
export default initialState;
