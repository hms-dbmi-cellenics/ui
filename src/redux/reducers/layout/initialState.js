const initialLayoutSpatial = {
  windows: {
    direction: 'row',
    first: {
      first: {
        first: 'UMAP', second: 'Spatial', direction: 'row', splitPercentage: 62,
      },
      second: {
        first: 'Heatmap',
        second: 'Cell sets and Metadata',
        direction: 'row',
        splitPercentage: 62,
      },
      direction: 'column',
      splitPercentage: 45,
    },
    second: 'Genes',
    splitPercentage: 70,
  },
  panel: 'Gene list',
};

const initialLayoutSingleCell = {
  windows: {
    direction: 'row',
    first: {
      first: {
        first: 'UMAP', second: 'Cell sets and Metadata', direction: 'row', splitPercentage: 62,
      },
      second: 'Heatmap',
      direction: 'column',
      splitPercentage: 45,
    },
    second: 'Genes',
    splitPercentage: 70,
  },
  panel: 'Gene list',
};

export { initialLayoutSpatial, initialLayoutSingleCell };
