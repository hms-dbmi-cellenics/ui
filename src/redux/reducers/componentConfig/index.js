import {
  UPDATE_CONFIG, CONFIGS_REPLACED,
  SAVE_CONFIG, LOAD_CONFIG, RESET_CONFIG,
  PLOT_DATA_LOADED, PLOT_DATA_LOADING, PLOT_DATA_ERROR, TRAJECTORY_NODES_SELECTION_UPDATED,
} from 'redux/actionTypes/componentConfig';

import { CELL_CLASS_DELETE } from 'redux/actionTypes/cellSets';

import initialState from 'redux/reducers/componentConfig/initialState';
import loadConfig from 'redux/reducers/componentConfig/loadConfig';
import updateConfig from 'redux/reducers/componentConfig/updateConfig';
import configsReplaced from 'redux/reducers/componentConfig/configsReplaced';
import saveConfig from 'redux/reducers/componentConfig/saveConfig';
import resetConfig from 'redux/reducers/componentConfig/resetConfig';
import plotDataLoaded from 'redux/reducers/componentConfig/plotDataLoaded';
import plotDataLoading from 'redux/reducers/componentConfig/plotDataLoading';
import plotDataError from 'redux/reducers/componentConfig/plotDataError';
import trajectoryNodesUpdated from 'redux/reducers/componentConfig/trajectoryNodesUpdated';

import cellClassDelete from 'redux/reducers/componentConfig/cellClassDelete';

const plotsReducer = (state = initialState, action) => {
  switch (action.type) {
    case LOAD_CONFIG:
      return loadConfig(state, action);
    case UPDATE_CONFIG:
      return updateConfig(state, action);
    case CONFIGS_REPLACED:
      return configsReplaced(state, action);
    case RESET_CONFIG:
      return resetConfig(state, action);
    case SAVE_CONFIG:
      return saveConfig(state, action);
    case PLOT_DATA_LOADED:
      return plotDataLoaded(state, action);
    case PLOT_DATA_LOADING:
      return plotDataLoading(state, action);
    case PLOT_DATA_ERROR:
      return plotDataError(state, action);
    case TRAJECTORY_NODES_SELECTION_UPDATED:
      return trajectoryNodesUpdated(state, action);
    case CELL_CLASS_DELETE: {
      return cellClassDelete(state, action);
    }
    default:
      return state;
  }
};

export default plotsReducer;
