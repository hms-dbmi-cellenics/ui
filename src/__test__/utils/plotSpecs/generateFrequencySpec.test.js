import { generateData } from 'utils/plotSpecs/generateFrequencySpec';
import { mockCellSets } from '__test__/test-utils/cellSets.mock';
import { initialPlotConfigStates } from 'redux/reducers/componentConfig/initialState';

describe('Generate frequency plot data', () => {
  const { hierarchy, properties } = mockCellSets;
  const config = initialPlotConfigStates.frequency;

  it('Generates proportional data', () => {
    const data = generateData(hierarchy, properties, config);

    expect(data).toMatchSnapshot();
  });

  it('Generates count data', () => {
    config.frequencyType = 'count';
    const data = generateData(hierarchy, properties, config);

    expect(data).toMatchSnapshot();
  });
});
