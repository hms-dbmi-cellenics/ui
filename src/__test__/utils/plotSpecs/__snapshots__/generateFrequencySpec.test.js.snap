// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Generate frequency plot data Generates count data 1`] = `
Object {
  "plotData": Array [
    Object {
      "color": "#01FFFF",
      "x": "sample-1",
      "y": 2,
      "yCellSetKey": "cluster-a",
    },
    Object {
      "color": "#23FFFF",
      "x": "sample-1",
      "y": 1,
      "yCellSetKey": "cluster-b",
    },
    Object {
      "color": "#45FFFF",
      "x": "sample-1",
      "y": 0,
      "yCellSetKey": "cluster-c",
    },
    Object {
      "color": "#01FFFF",
      "x": "sample-2",
      "y": 0,
      "yCellSetKey": "cluster-a",
    },
    Object {
      "color": "#23FFFF",
      "x": "sample-2",
      "y": 1,
      "yCellSetKey": "cluster-b",
    },
    Object {
      "color": "#45FFFF",
      "x": "sample-2",
      "y": 2,
      "yCellSetKey": "cluster-c",
    },
  ],
  "xNamesToDisplay": Array [
    "Sample 1",
    "Sample 2",
  ],
  "yNamesToDisplay": Array [
    "cluster a",
    "cluster b",
    "cluster c",
  ],
}
`;

exports[`Generate frequency plot data Generates proportional data 1`] = `
Object {
  "plotData": Array [
    Object {
      "color": "#01FFFF",
      "x": "sample-1",
      "y": "66.667",
      "yCellSetKey": "cluster-a",
    },
    Object {
      "color": "#23FFFF",
      "x": "sample-1",
      "y": "33.333",
      "yCellSetKey": "cluster-b",
    },
    Object {
      "color": "#45FFFF",
      "x": "sample-1",
      "y": "0.000",
      "yCellSetKey": "cluster-c",
    },
    Object {
      "color": "#01FFFF",
      "x": "sample-2",
      "y": "0.000",
      "yCellSetKey": "cluster-a",
    },
    Object {
      "color": "#23FFFF",
      "x": "sample-2",
      "y": "33.333",
      "yCellSetKey": "cluster-b",
    },
    Object {
      "color": "#45FFFF",
      "x": "sample-2",
      "y": "66.667",
      "yCellSetKey": "cluster-c",
    },
  ],
  "xNamesToDisplay": Array [
    "Sample 1",
    "Sample 2",
  ],
  "yNamesToDisplay": Array [
    "cluster a",
    "cluster b",
    "cluster c",
  ],
}
`;
